### PR TITLE
Host blog on Heroku

### PR DESCRIPTION
**Pull request by jingweno**
Moving https://owenou.com from GitHub Pages to Heroku and caching assets with Fastly.

**How it works**

The Jekyll static site is compiled as part of the [heroku-buildpack-ruby's asset precompile](https://github.com/heroku/heroku-buildpack-ruby/blob/master/lib/language_pack/ruby.rb#L789-L803) and is served with [heroku-buildpack-static](https://github.com/hone/heroku-buildpack-static) using Nginx. Assets are cached with Fastly and cache is purged on each Heroku push at asset precompilation. Asset URLs are rewritten to point to Fastly CDN by [jekyll-assets](https://github.com/jekyll/jekyll-assets#configuration).

Alternatively, to maximize performance, Fastly can be put in front the Heroku Dyno to serve everything: DNS (`owenou.com`) -> Fastly CDN edge (`owenou-herokuapp-com.global.ssl.fastly.net`) -> Heroku Dyno. But [setting up a custom cert on Fastly is expensive](https://www.fastly.com/pricing) so in the case of serving a blog, only caching assets with Fastly is sufficient.

**Advantages**

* HTTPS support. This has been [a long time request](https://github.com/isaacs/github/issues/156) for GitHub Pages.
* Caching with CDN. Thanks to `heroku-buildpack-static` [cache headers can be easily defined](https://github.com/hone/heroku-buildpack-static#custom-headers).

Now my blog is very fast(ly) :smile_cat:!

Fastly serves everything:

screen shot 2015-11-22 at 8 51 13 am

Fastly only serves assets:

screen shot 2015-11-22 at 12 23 01 pm

**Disadvantages**

* Cost. Heroku Free Dyno ($0) + Heroku SSL Domain ($20) + Fastly Fast Plan ($50) = $70, while GitHub Pages is free...It will be even more expensive to have custom cert on Fastly. However, it can be totally free on Heroku for casual use without custom SSL domain and Fastly.

**Wish List**

* A generator to set things up, e.g., `heroku static:init --jekyll`
* [Heroku Deploy Hooks](https://devcenter.heroku.com/articles/deploy-hooks) can run a script in the Dyno so that purging Fastly cache is easier. Currently I'm [putting this in the `assets:precompile` step](https://github.com/jingweno/jingweno.github.io/blob/heroku/Rakefile#L9-L14) as part of the Heroku Ruby buildpack.

### CODE REVIEW

**Comment by jingweno** [comment]
I just realized it's pretty expensive to set up a custom cert on Fastly: https://www.fastly.com/pricing. So it may not worth it for a blog...

**Comment by mikehale** [comment]
Very cool. Looks like with fastly total load time is < 1000ms.

**Comment by jingweno** [comment]
@mikehale Yeah, from my benchmarking, all assets downloads finished < 500ms (the rest is DOM load time).

File: Procfile [hunk 1]
```diff
@@ -0,0 +1 @@
+web: bin/boot
```
Comment (atmos) [inline Procfile]: Where does this executable come from?

Comment (jingweno) [inline Procfile]: It's from [heroku-buildpack-static](https://github.com/hone/heroku-buildpack-static/blob/master/scripts/boot).

Comment (atmos) [inline Procfile]: I switched over to a letsencrypt cert and got away with out using the Procfile somehow. https://github.com/atmos/atmos.github.io/pull/9

Comment (jingweno) [inline Procfile]: Hmm...interesting. Maybe @hone could comment on why this would work.

Comment (jingweno) [inline Procfile]: @atmos BTW, very cool of using letsencrypt!

**Pull request merged**